python/src/main/resources/python/zeppelin_context.py

```diff
@@ -158,13 +159,12 @@ def show_dataframe(self, df, show_index=False, **kwargs):
                 body_buf.write("\t")
                 body_buf.write(str(cell))
             body_buf.write("\n")
-        body_buf.seek(0); header_buf.seek(0)
-        #TODO(bzz): fix it, so it shows red notice, as in Spark
-        print("%table " + header_buf.read() + body_buf.read()) # +
-        #      ("\n<font color=red>Results are limited by {}.</font>" \
-        #          .format(self.max_result) if limit else "")
-        #)
+        body_buf.seek(0)
+        header_buf.seek(0)
+        print("%table " + header_buf.read() + body_buf.read())
         body_buf.close(); header_buf.close()
+        if exceed_limit:
+            print("%html <font color=red>Results are limited by {}.</font>".format(self.max_result))
```

Comment:
do we need a newline separating %html from before that?

Reply:
Not necessary, frontend will separate them correctly.